library/core/src/panicking.rs

```diff
@@ -219,7 +219,8 @@ fn assert_failed_inner(
         Some(args) => panic!(
             r#"assertion failed: `(left {} right)`
   left: `{:?}`,
- right: `{:?}`: {}"#,
+ right: `{:?}`,
+context: `{:?}`"#,
```

Comment:
It looks like the intent here was to align left: and right: with each other along the `:` - probably we want to do the same with context: if we do this?

Reply:
updated as suggested

Reply:
Do I continue to submit more commits according to yaahc's suggestions, or close this PR?

Reply:
Go ahead and update the format as suggested.

Reply:
Change `panicking.rs` in core package and one in std remains the same as the latter makes more failed tests.
The current output format is like this:
```
thread 'main' panicked at 'assertion failed: `(upper_bounds == target)`
   Error: `1 + 1 definitely should be 3`,
   upper_bounds: `2`,
   target: `3`', /home/kougami/temp/rust/src/test/ui/macros/assert-eq-macro-msg.rs:8:5
```

Reply:
Oh, I didn't mean to hard code `upper_bound` and `target` as the names of the values, I had assumed you'd be able to stringify the actual original names and print those, but for now I think just leave it as `left` and `right` since I have no idea how difficult it would be to make the change to use the stringified original names.

Reply:
This is my mistake. Do you want to stringify values instead of outputting parameter names. @yaahc

Reply:
not exactly, `stringify!` stringifies exactly whatever you pass into it, so calling `stringify!(op)` will just give you "op". Instead you'd need to call `stringify!` from inside of the `assert_eq!` macro_rules definition where you still have access to the original tokens, then you'd have to add arguments to `assert_failed` and `assert_failed_inner` to pass the strings all the way to the inner function that formats them, so it would look something like this:

```rust
// in `library/core/src/macros/mod.rs`
macro_rules! assert_eq {
    ($left:expr, $right:expr $(,)?) => ({
        match (&$left, &$right) {
            (left_val, right_val) => {
                if !(*left_val == *right_val) {
                    let kind = $crate::panicking::AssertKind::Eq;
                    let left_name = stringify!($left);
                    let right_name = stringify!($right);
                    // The reborrows below are intentional. Without them, the stack slot for the
                    // borrow is initialized even before the values are compared, leading to a
                    // noticeable slow down.
                    $crate::panicking::assert_failed(kind, &*left_val, &*right_val, left_name, right_name, $crate::option::Option::None);
                }
            }
        }
    });
   // the rest of the assert_eq definition ...
}

// in `library/core/src/panicking` assert_failed_inner
fn assert_failed_inner(
    kind: AssertKind,
    left: &dyn fmt::Debug,
    right: &dyn fmt::Debug,
    left_name: &'static str,
    right_name: &'static str,
    args: Option<fmt::Arguments<'_>>,
) -> ! {
    let op = match kind {
        AssertKind::Eq => "==",
        AssertKind::Ne => "!=",
        AssertKind::Match => "matches",
    };

    match args {
        Some(args) => panic!(
            r#"assertion failed: `({left_name} {} {right_name})`
  {left_name}: `{:?}`,
 {right_name}: `{:?}`: {}"#,
            op, left, right, args
        ),
       // ....
}
```

Reply:
I tried my best to find some code snippets that needs to be updated. Is there anything else I'm missing?

Reply:
no, this looks perfect! We just need to get the exact output sorted and update the test UI files so everything passes. Right now looking at the failed test in CI the output looks like this:

```
 thread 'main' panicked at 'assertion failed: `(1 + 1 == 5)`
  1 + 1: `2`,
 5: `5`', /checkout/src/test/ui/test-attrs/test-panic-abort.rs:34:5
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```

So we probably want to clean this up a bit, and we may want to rethink how many times we repeat the stringified version of the expression. Here's my suggestion for what we should aim for:

```
thread 'main' panicked at 'assertion failed: `(1 + 1 == 5)`
  left: `2`,
 right: `5`'
    at: /checkout/src/test/ui/test-attrs/test-panic-abort.rs:34:5
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```

Reply:
updated my commit